### ISSUE TITLE
Path Parameter added in C-Sharp Generator

### ISSUE THREAD

**Issue by jhitchcock** [swagger-api/swagger-codegen#1332]
Within the C sharp generate the line 
pathParams.Add("format", "json");

exists in api.moustache 

Not sure why its there, But I also noticed, Accept header is not set in the requests

**Comment by wing328** [comment]
@jhitchcock For `pathParams.Add("format", "json");` the logic is already there before I started making contribution to the C# code generator (and other components of the project). My understand is that it automatically replaces `/pet/{petId}.{format}` with `/pet/{petId}.json` for example (since the code generator can only deserialize JSON response into models at the moment).

Is this logic causing issue to your use case (e.g. you already have a path parameter named `format` in the URL) ?

For Accept header, it should be determined automatically by RestSharp if I'm not mistaken. If it's also causing issue to your use case, please provide the swagger spec for reproducing the issue.

**Comment by jhitchcock** [comment]
Hi, Thanks for the fast reply. Thanks for the explanation on the first one, I wasn't thinking straight and didn't read it as PATH param! 

Re the application/json Accept header. My API will take application/json or application/xml as the accept header, but I only want to use the JSON representations. in Swagger UI the application/json accept header is sent in the request, but it doesn't seem to be using the SDK. Ive added 

"produces":[
               "application/json"

Into my Swagger JSON, but form debugging, XML is what's coming back. 

Basically, I want the SDK to send that accept header, without adding it explicitly as a parameter to each end point. 

For now, I have hard coded it in the c sharp generator Java file, and this seems to fix the issue. Though, Probably not the right way to go about it, It's got me up and running for now!

**Comment by wing328** [comment]
@jhitchcock `produces` should be used similar to what we've done for other languages. Hopefully I should be able to file a PR to address that in the coming week.

**Comment by wing328** [comment]
@jhitchcock I've submitted #1333 to address the issue. I wonder if you can perform a test to confirm the fix works for you.

**Comment by wing328** [comment]
@jhitchcock PR merged into master. Let me know if you still see issue with "Accept" header.

**Comment by jhitchcock** [comment]
Thanks so much. Just re-build with the latest version, and it's working perfectly without my 'hack' 

:)
